### PR TITLE
Removing html angular extractor for issue 4018

### PR DESCRIPTION
**Pull request by zashary**
### Description
This PR removes the angular html extractor for dashboards and is a part of the larger de-angular effort.

 Part of work to address https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4018

### Issues Resolved

* Removes angular html extractor
* https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4018

### Check List

- [ ] All tests pass
  - [x] `yarn test:jest`
  - [x] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: packages/osd-config-schema/src/errors/__snapshots__/schema_error.test.ts.snap
Comment (ashwin-pc) [inline packages/osd-config-schema/src/errors/__snapshots__/schema_error.test.ts.snap:1]: Looks like this change is causing the unit test to fail. reverting it should make your tests pass.

**Review by ashwin-pc — COMMENTED** [review]
The change looks good to me :) can you make sure that you sign the commit and add a changelog entry for the PR and revert the change to the `schema_error.test.ts` snapshot? With that i think we can merge this PR while you work on deangularizing `i18n`

File: src/dev/i18n/extract_default_translations.js [hunk 1]
```diff
@@ -81,33 +81,31 @@ export async function matchEntriesWithExctractors(inputPath, options = {}) {
...
       const resolvedPath = path.resolve(inputPath, entry);
 
-      if (resolvedPath.endsWith('.html')) {
-        paths.htmlEntries.push(resolvedPath);
-      } else {
-        paths.codeEntries.push(resolvedPath);
-      }
+      paths.codeEntries.push(resolvedPath);
+
+      // if (resolvedPath.endsWith('.html')) {
```
Comment (ananzh) [inline src/dev/i18n/extract_default_translations.js:98]: could just remove this part.

Comment (ananzh) [reply src/dev/i18n/extract_default_translations.js:98]: nit: should we do 

```
const codeEntries = entries.reduce(
    (paths, entry) => {
      const resolvedPath = path.resolve(inputPath, entry);

      paths.codeEntries.push(resolvedPath);
      return paths;
      },
     [] 
  );
```
?

File: src/dev/i18n/extractors/index.js [hunk 2]
```diff
@@ -29,4 +29,4 @@
  */
 
 export { extractCodeMessages } from './code';
-export { extractHtmlMessages } from './html';
+//export { extractHtmlMessages } from './html';
```
Comment (ananzh) [inline src/dev/i18n/extractors/index.js:32]: should we just remove this line?

File: packages/osd-config-schema/src/errors/__snapshots__/schema_error.test.ts.snap
Comment (ananzh) [reply packages/osd-config-schema/src/errors/__snapshots__/schema_error.test.ts.snap:1]: For sign the commit in the future, you could do
```
git config --global user.name "Your name" 
git config --global user.email your@email.com
```

Then for commit: `git commit -sm "<commit msg>"`

For fixing this one, you could amend the commit directly by adding the sign-off. For example
```
git commit --amend
```
Then add 
```
Signed-off-by: zashary <your email>
```

Comment (ananzh) [reply packages/osd-config-schema/src/errors/__snapshots__/schema_error.test.ts.snap:1]: For Changelog, we mean add this PR in CHANGELOG.md. You could check other PRs for a reference.

File: config/opensearch_dashboards.yml [hunk 3]
```diff
@@ -135,6 +135,7 @@
 # Specifies locale to be used for all localizable strings, dates and number formats.
 # Supported languages are the following: English - en , by default , Chinese - zh-CN .
 #i18n.locale: "en"
+i18n.locale: "es"
```
[the remark this answers is not in the document]
Comment (zashary) [reply config/opensearch_dashboards.yml:138]: This was part of testing. I've gone ahead and removed it.

File: src/dev/i18n/extract_default_translations.js (hunk 1 above)
Comment (zashary) [reply src/dev/i18n/extract_default_translations.js]: That's a good call out and looks a lot cleaner that way! I've gone ahead and updated this bit.

**Review by ashwin-pc — APPROVED** [review]
LGTM :) Thanks for the contribution @zashary

**Review by ananzh — APPROVED** [review]

**Pull request merged**